### ISSUE TITLE
Unable to checkout branch when using git worktree

### ISSUE THREAD

**Issue by fatih-asici** [agateau/git-bonsai#5]
I am using "git worktree" to checkout branches into multiple directories. git bonsai fails with an output like this:

```
...
Info: Updating v0.7-branch
fatal: 'v0.7-branch' is already checked out at '/home/user/git/project-0.7'

Error: Failed to checkout branch
```

**Comment by agateau** [comment]
Hi Fatih, thanks for the feedback (and sorry for the late answer). I'd like to look into this, how do I setup a "git worktree"?

**Comment by fatih-asici** [comment]
To add a new worktree which the branch "foo" is checked out in:

```bash
git worktree /path/to/a/new/directory foo
```

That's enough to setup. No other initial setup is necessary.

If you run git-bonsai in this repository (doesn't matter if it is main tree or worktree), it will fail.

**Comment by agateau** [comment]
I just tried it and it seems to work for me, as long as I don't try to delete the branch checked out in a separate worktree.

Here is what I did:

First I created a test repository with the [create-demo-repository](https://github.com/agateau/git-bonsai/blob/7387bbf47e90bac5e0f1e384e4589f81de3e5c90/create-demo-repository) script:

```
$ ./create-demo-repository 
Initialized empty Git repository in /home/perso/src/git-bonsai/demo/.git/
[master (root-commit) 62361ef] Init
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 README
Switched to a new branch 'topic1'
[topic1 caaf36b] Create topic1
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1
Switched to a new branch 'topic1-1'
[topic1-1 b96415a] Create topic1-1
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1-1
Switched to a new branch 'topic2'
[topic2 66806e5] Create topic2
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic2
Switched to branch 'topic1'
Merge made by the 'recursive' strategy.
 topic1-1 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1-1
Switched to branch 'master'
Merge made by the 'recursive' strategy.
 topic1   | 0
 topic1-1 | 0
 2 files changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1
 create mode 100644 topic1-1
Switched to a new branch 'duplicate1'
[duplicate1 3cc6af2] Create duplicate1
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 duplicate1
Switched to branch 'master'
```

The demo repository looks like this:
```
$ cd demo
$ git branch
  duplicate1
  duplicate2
* master
  topic1
  topic1-1
  topic2

$ git log --graph --oneline
*   aca6b86 (HEAD -> master) Merging topic1
|\  
| *   f42bdf7 Merging topic1-1
| |\  
| | * b96415a Create topic1-1
| |/  
| * caaf36b Create topic1
|/  
* 62361ef Init
```

Then I created a worktree for the `duplicate1` branch:

```
$ git worktree add ../duplicate1 duplicate1
Preparing worktree (checking out 'duplicate1')
HEAD is now at 3cc6af2 Create duplicate1
```

And ran git-bonsai in the main tree:

```
$ git bonsai
Info: Fetching changes
These branches point to the same commit, but no other branch contains this commit, so you can delete all of them but one.

Select branches to delete: duplicate2
Info: Deleting duplicate2
Select branches to delete: topic1, contained in:
      - duplicate1
      - master 
, topic1-1, contained in:
      - topic1
      - duplicate1
      - master 

Info: Deleting topic1
Info: Deleting topic1-1
```

Running it in the worktree is OK as well. Telling git bonsai to delete the `duplicate1` branch fails, as expected.

Is there anything else I need to do to reproduce the issue? Which version of git-bonsai do you use?

**Comment by fatih-asici** [comment]
Using version 0.2.0. In order to reproduce there should be a remote tracking branch:

```sh
$ ./create-demo-repository 
Initialized empty Git repository in /home/fatih/git/git-bonsai/demo/.git/
[master (root-commit) ffb8f57] Init
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 README
Switched to a new branch 'topic1'
[topic1 3f8680d] Create topic1
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1
Switched to a new branch 'topic1-1'
[topic1-1 b78458a] Create topic1-1
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1-1
Switched to a new branch 'topic2'
[topic2 55d6680] Create topic2
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic2
Switched to branch 'topic1'
Merge made by the 'recursive' strategy.
 topic1-1 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1-1
Switched to branch 'master'
Merge made by the 'recursive' strategy.
 topic1   | 0
 topic1-1 | 0
 2 files changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 topic1
 create mode 100644 topic1-1
Switched to a new branch 'duplicate1'
[duplicate1 fa3cc48] Create duplicate1
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 duplicate1
Switched to branch 'master'

$ git clone demo demo2
Cloning into 'demo2'...
done.

$ cd demo2/
$ git worktree add ../duplicate2
Preparing worktree (new branch 'duplicate2')
HEAD is now at eff14eb Merging topic1

$ cd ../duplicate2/
$ git checkout topic1 --
Branch 'topic1' set up to track remote branch 'topic1' from 'origin'.
Switched to a new branch 'topic1'

$ cd ../demo2/
$ git bonsai 
Info: Fetching changes
Info: Updating master
Already up to date.

Info: Updating topic1
fatal: 'topic1' is already checked out at '/home/fatih/git/git-bonsai/duplicate2'

Error: Failed to checkout branch

```

**Comment by agateau** [comment]
Oh right! I can reproduce the bug now. Thanks for your feedback, going to look into it.

**Comment by agateau** [comment]
I just merged in a first approach to this, which just skips worktree branches. That's not ideal because you still have to run git bonsai in all your worktrees, but at least it does not crash anymore.

You can try it out, either by building it yourself or using a 0.2.1 binary from https://builds.agateau.com/git-bonsai/

**Comment by fatih-asici** [comment]
Tested with 0.2.1 binary. It works now. Thanks for the fix.

**Comment by agateau** [comment]
I spent a bit of time playing with the idea of supporting multiple worktrees, but I am worried it's going to make git bonsai too complicated: an operation like a branch update can be applied to different worktrees, but removing dead branches is something I feel should not be applied everywhere: it would be scary if git bonsai removed a worktree because its branch has been merged in another one.

I think it's wiser to keep it simple and just skip worktree branches. At least for now.
